### PR TITLE
[Execution] Add simple check to commit_blocks_ext.

### PR DESCRIPTION
**Pull request by JoshLind**
### Description

Add an assertion to verify that the last block id matches the block id to commit.

### Test Plan
Existing tests.

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/3690)

### CODE REVIEW

File: execution/executor/src/block_executor.rs [hunk 1]
```diff
@@ -220,19 +220,35 @@ where
...
         );
+        let last_block_id = *block_ids.last().unwrap();
+        if last_block_id != block_id_to_commit {
+            return Err(anyhow::anyhow!(
+                "Block id to commit ({:?}) does not match last block id ({:?})!",
+                block_id_to_commit,
+                last_block_id
+            )
+            .into());
+        }
```
Comment (msmouse) [inline execution/executor/src/block_executor.rs:251]: I'm not exactly sure how it works with reconfig suffixes @zekun000 to decide.

Comment (zekun000) [reply execution/executor/src/block_executor.rs:251]: this block id still matches, those checks probably should be assertion than error, it's just invariant

File: execution/executor/src/block_executor.rs [hunk 2]
```diff
@@ -220,19 +220,35 @@ where
...
+        if block_ids.is_empty() {
+            return Err(anyhow::anyhow!("Cannot commit 0 blocks!").into());
+        }
+
+        // Check for any potential retries
         let committed_block = self.block_tree.root_block();
         if committed_block.num_persisted_transactions()
             == ledger_info_with_sigs.ledger_info().version() + 1
         {
-            // a retry
```
Comment (zekun000) [inline execution/executor/src/block_executor.rs:227]: why?

File: execution/executor/src/block_executor.rs (hunk 1 above)
Comment (msmouse) [reply execution/executor/src/block_executor.rs:251]: K. Then I agree, I'd rather panic here.

File: execution/executor/src/block_executor.rs (hunk 2 above)
Comment (JoshLind) [reply execution/executor/src/block_executor.rs:227]: Comment was moved above: `// Check for any potential retries`. Let me know if you want both.

File: execution/executor/src/block_executor.rs (hunk 1 above)
Comment (JoshLind) [reply execution/executor/src/block_executor.rs]: Cool, I've updated it to panic! 😄

**Review by davidiw — APPROVED** [review]

**Pull request merged**